### PR TITLE
DNEXT-23501:add the new api SECValidateAccessToken in docs.

### PR DESCRIPTION
**Pull request by shaiknayabrasool-s**

### CODE REVIEW

File: docs/reference/Func/SECValidateAccessToken.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Output :
+(routine)  -  Common return values:
+  * ERR_BSAFE_NULLPARAM - Invalid NULL inputs or no trusted OIDC providers
+  * ERR_BSAFE_NON_EXISTENT - OIDC Provider not initialized or cannot be used for bearerAuth
+  * ERR_BSAFE_TOOSMALL - Email address larger than dwMaxEmailSize
+  * ERR_BSAFE_BAD_ATTRIBUTES - JWT lacking mandatory attributes or issuer/aud/scope/azp not matching
+  * ERR_BSAFE_CERT_VALIDITY - JWT issued in the future
+  * ERR_SECURE_EXPIRED_CERT - JWT expired in the past
+  * ERR_BSAFE_BAD_SIGNATURE - Invalid signature on JWT
+  * ERR_BSAFE_BAD_OPCODE - Unsupported combination of signing algorithm and key
```
Comment (saravananbalakris) [inline docs/reference/Func/SECValidateAccessToken.md:53]: On Success what is returned from this function?

File: docs/reference/Symb/ERR_BSAFE_XXX.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	ERR_BSAFE_NON_EXISTENT	  -  OIDC Provider not initialized or cannot be used for bearerAuth .
+
+	ERR_BSAFE_TOOSMALL	  -  Email address larger than dwMaxEmailSize.
+
+	ERR_BSAFE_BAD_ATTRIBUTES	  -  JWT lacking mandatory attributes or issuer/aud/scope/azp not matching.
+
+	ERR_BSAFE_CERT_VALIDITY	  -  JWT issued in the future.
+
+	ERR_SECURE_EXPIRED_CERT	  -  JWT expired in the past .
```
Comment (saravananbalakris) [inline docs/reference/Symb/ERR_BSAFE_XXX.md]: Why you add ERR_SECURE_xxx as part of ERR_BSAFE_xxx

Comment (saravananbalakris) [reply docs/reference/Symb/ERR_BSAFE_XXX.md]: Is errors are considered as symbols ?

File: docs/reference/Func/SECValidateAccessToken.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	DWORD  dwFlags,
+	void  *vpOptionalParams,
+	DWORD  dwMaxEmailSize,
+	char  *retszEmail,
+	DWORD  *retdwDurationSec);
+```
+
+**Description :**
+
+Validates a signed JWT access token that was generated by a trusted OIDC provider. Intended to be called by multiple different server tasks, including external code.
```
Comment (saravananbalakris) [inline docs/reference/Func/SECValidateAccessToken.md:21]: What is JWT & OIDC stands for ?

File: docs/reference/Data/JWT_VALIDATE_OPTIONAL_PARAMS.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+**Description :**
+
+Validates a signed JWT access token.<br>
+<font color="#121212" face="Arial">#define fJWT_validate_AllowExpired 0x00000001<br>
+#define fJWT_validate_AllowMSWorkarounds 0x00000002<br>
+#define fJWT_validate_UseCustomEmailClaim 0x00000010<br>
+#define fJWT_validate_AllowAlternateAud 0x00000020<br>
+#define fJWT_validate_EnforceAllowedClients 0x00000040</font><font size="4"> </font>
```
Comment (saravananbalakris) [inline docs/reference/Data/JWT_VALIDATE_OPTIONAL_PARAMS.md]: Is these fJWT_validate_xxx shall be symbols?

**Review by saravananbalakris — CHANGES_REQUESTED** [review]
Clarifications

File: docs/reference/Data/JWT_VALIDATE_OPTIONAL_PARAMS.md (hunk 4 above)
Comment (shaiknayabrasool-s) [reply docs/reference/Data/JWT_VALIDATE_OPTIONAL_PARAMS.md]: NO, this define values we aren't consider as Symbols.

File: docs/reference/Func/SECValidateAccessToken.md (hunk 3 above)
Comment (shaiknayabrasool-s) [reply docs/reference/Func/SECValidateAccessToken.md:21]: JWT - JSON Web Token
OIDC - Open ID Connect

File: docs/reference/Symb/ERR_BSAFE_XXX.md (hunk 2 above)
Comment (shaiknayabrasool-s) [reply docs/reference/Symb/ERR_BSAFE_XXX.md]: Yes, we can consider as a Symbols and, can create ERR_SECURE_XXX as an alone Symbol.

File: docs/reference/Func/SECValidateAccessToken.md (hunk 1 above)
Comment (shaiknayabrasool-s) [reply docs/reference/Func/SECValidateAccessToken.md:53]: *retpEmail        - On success, filled in with email address from JWT.
*retqwDurationSec - Optional. On success, filled in with seconds left before expiration.

Comment (erika-dg) [reply docs/reference/Func/SECValidateAccessToken.md:53]: @shaiknayabrasool-s Line 32: pszResourceURL  -  Expected value in audience (aud) claim, such **ashttps://www.example.com** - can we add a space after "as" here?

Comment (shaiknayabrasool-s) [reply docs/reference/Func/SECValidateAccessToken.md:53]: Corrected.

**Review by raghu-madat — APPROVED** [review]

**Review by raghu-madat — APPROVED** [review]

File: docs/reference/Func/SECValidateAccessToken.md (hunk 1 above)
Comment (saravananbalakris) [reply docs/reference/Func/SECValidateAccessToken.md:53]: On Success, it shall return NOERROR, that's my understanding

**Review by saravananbalakris — APPROVED** [review]

**Pull request merged**